entity/contents/snapshot.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/macros.h"
+#include "impeller/geometry/matrix.h"
+#include "impeller/geometry/rect.h"
+#include "impeller/renderer/texture.h"
+
+namespace impeller {
+
+class ContentContext;
+class Entity;
+
+/// Represents a screen space texture and its intended draw position.
+struct Snapshot {
+  std::shared_ptr<Texture> texture;
+  /// The offset from the origin where this texture is intended to be
+  /// rendered.
+  Vector2 position;
```

Comment:
Can this be a Matrix instead? Seems that would add additional utility. You're need to invert it to apply the entity transformation.

Reply:
Yup! This was something I was planning on doing in a follow-up if you don't mind. I'm planning to make FilterContents smarter about producing UVs for given FilterInputs, which will remove unnecessary subpasses for non-contents/texture inputs. Having a local transform here will be part of that.